src/Microsoft.AspNetCore.Mvc.Core/ModelBinding/Binders/SimpleTypeModelBinder.cs

```diff
@@ -47,23 +47,30 @@ public Task BindModelAsync(ModelBindingContext bindingContext)
             {
                 var value = valueProviderResult.FirstValue;
 
-                object model = null;
-                if (!string.IsNullOrWhiteSpace(value))
-                {
-                    model = _typeConverter.ConvertFrom(
-                        context: null,
-                        culture: valueProviderResult.Culture,
-                        value: value);
-                }
-
+                object model;
                 if (bindingContext.ModelType == typeof(string))
                 {
-                    var modelAsString = model as string;
-                    if (bindingContext.ModelMetadata.ConvertEmptyStringToNull &&
-                        string.IsNullOrEmpty(modelAsString))
+                    // Already have a string. No further conversion required but handle ConvertEmptyStringToNull.
+                    if (bindingContext.ModelMetadata.ConvertEmptyStringToNull && string.IsNullOrWhiteSpace(value))
                     {
                         model = null;
                     }
+                    else
+                    {
+                        model = value;
+                    }
+                }
+                else if (string.IsNullOrWhiteSpace(value))
+                {
+                    // Other than the StringConverter, converters Trim() the value then throw if the result is empty.
```

Comment:
:see_no_evil: Agree there'd be no reason to document anything here if our reasoning stopped at "don't call type converters w/ whitespace (or null)". But it's a bit deeper than that.